modules/apps/transfer/types/trace.go

```diff
@@ -32,9 +37,25 @@ func ParseDenomTrace(rawDenom string) DenomTrace {
 		}
 	}
 
+	path := []string{}
+	baseDenom := []string{}
+	length := len(denomSplit)
+	r, err := regexp.Compile(fmt.Sprintf("%s[0-9]+", channeltypes.ChannelPrefix))
+	if err != nil {
+		panic("could not compile regexp 'channel-[0-9]+'")
+	}
+	for i := 0; i < length; i = i + 2 {
+		if i < length-1 && denomSplit[i] == PortID && r.MatchString(denomSplit[i+1]) {
```

Comment:
Checking that both the port ID is `transfer` and the channel ID matches the regex `channel-[0-9]+` might be too restrictive, right? Because the specs don't force the implementations to follow these two patterns, right? 
If I remember correctly the transfer module of Agoric actually uses a custom port ID (at least it did back in September). And other implementations of IBC might also decide to use a different pattern for the channel identifiers. And don't get me started if a custom transfer module decides to use slashes in the port ID...
So it seems like it's a bit tricky to parse the denom and be able to cover all these possible scenarios.

Reply:
We should check that the portID is transfer. Since this will always pass through the transfer module and even if tokens came from a previous app, we care about the chain of `transfer` module hops. But we do not enforce channel-id naming.

Reply:
So keep transfer and remove regex for channelIDs

Reply:
Done.